frontend/catalyst/pennylane_extensions.py

```diff
@@ -191,12 +191,22 @@ def _make_jaxpr_check_differentiable(f: Differentiable, grad_params: GradParams,
                     return_ops.append(eq.primitive)
                     break
 
-        if method == "ps" and any(prim not in [expval_p, probs_p] for prim in return_ops):
+        assert isinstance(
+            f, qml.QNode
+        ), "Differentiation methods other than finite-differences can only operate on a QNode"
```

Comment:
I think this should be an error, we only use assertions for "non-user facing" checks that would indicate a bug if triggered.

Reply:
Right, the assertion here is actually redundant because the check would be caught first by the `Grad.__init__` method on line 265 of this file. I added the assert because we later read the `.diff_method` property, which is only defined on a QNode. The assert here basically emulates a `cast<QNode>(f)` in C++.

Another option would just be to assume that the check on line 265 succeeded and thus we definitely have a QNode.

Reply:
Oh got it, if there is already a higher-level check and this is an internal assert than it's all good 👍